pwiz_tools/Skyline/Model/DocSettings/PeptideSettings.cs

```diff
@@ -685,10 +642,27 @@ public void ReadXml(XmlReader reader)
                 reader.ReadStartElement();
                 // Read child elements
                 RetentionTime = reader.DeserializeElement<RetentionTimeRegression>();
-                IonMobilityPredictor = reader.DeserializeElement<IonMobilityPredictor>();
+                if (reader.IsStartElement(DriftTimePredictor.EL.predict_drift_time)) // Reading an older format
+                    obsoleteDriftTimePredictor = reader.DeserializeElement<DriftTimePredictor>();
                 reader.ReadEndElement();                
             }
 
+            // If we're reading an older format, gather needed ion mobility info to pass along to TransitionSettings
+            if ((obsoleteDriftTimePredictor != null && !obsoleteDriftTimePredictor.IsEmpty) ||
+                !obsoleteSpectralLibraryIonMobilityWindowWidthCalculator.IsEmpty ||
+                obsoleteUseSpectralLibraryDriftTimes)
+            {
+                var dir =string.IsNullOrEmpty(reader.BaseURI) ? Directory.GetCurrentDirectory() : Path.GetDirectoryName(reader.BaseURI);
+                ObsoleteIonMobilityValues =
+                    (obsoleteDriftTimePredictor == null

```

Comment:
I'll see if I can configure ReSharper at the team level to stop suggesting conversion to ternaries. It's a distraction getting a blue squiggle if this is our preferred style.

Reply:
It's already at hint, so I guess it's actually green dots rather than blue squiggle. It just makes for a lot of green dots for something that usually seems to end up needing to be backed out. Seems like an unhelpful distraction.